### ISSUE TITLE
Integration tests issue

### ISSUE THREAD

**Issue by Alexander-Gruev** [luchob/softuni-sept-2021-misc#88]
Привет, Лъчо! Проблемите с тестовете се сипят като първия сняг.. 
Аз имам 17 теста, които, пуснати по отделно, минават всичките, но когато са всичките тест с име testPostRegisterShouldRegisterUserWithCorrectFields ми дава тази грешка: javax.persistence.NonUniqueResultException: query did not return a unique result: 6. Предполагам проблемът идва от общия контекст, но не разбирам какво да направя на практика

https://github.com/Alexander-Gruev/Fitshop

**Comment by Alexander-Gruev** [comment]
Да, аз съм използвал вариант едно, като във всеки от класовете за интеграционни тестове имам 
```
@AfterEach
    void tearDown() {
        userRepository.deleteAll();
    }
```
Но пак тестът си гърми

**Comment by Alexander-Gruev** [comment]
Оправих го - проблемът не беше, че userRepository връща повече от един резултат, а че userRoleRepository го прави

**Comment by luchob** [comment]
Йей, знаех си - то лесна работа било :P :-)
